src/sentry/utils/snql.py

```diff
@@ -1,50 +1,80 @@
 import random
-from typing import Optional, Set
+from typing import List, Mapping, NamedTuple, Optional, Set
 
 from sentry import options
 
-referrer_blacklist: Set[str] = set()
-referrers_by_entity = {
-    "api.performance.durationpercentilechart": "discover_transactions",
-    "api.performance.vital-detail": "discover_transactions",
-    "api.performance.status-breakdown": "discover_transactions",
-    "api.trends.get-percentage-change": "discover_transactions",
-    "api.performance.transaction-summary": "discover_transactions",
-}
-referrer_prefixes = ["outcomes.", "sessions.", "tsdb-modelid:6", "tsdb-modelid:5", "incidents."]
-referrers = {
-    "eventstore.get_events",
-    "group.filter_by_event_id",
-    "incidents.get_incident_aggregates",
-    "tagstore.get_tag_value_paginator_for_projects",
-    "api.serializer.projects.get_stats",
-    "tsdb-modelid:407",
-    "testing.test",
-}
-
-
-# Returns None (do not use SnQL), "auto" (use dataset as entity) or a specific entity to use.
-def should_use_snql(referrer: Optional[str]) -> Optional[str]:
-    if not referrer or referrer in referrer_blacklist:
+
+class SnQLOption(NamedTuple):
```

Comment:
I would probably go with `SnqlOption` or `SNQLOption` as the current casing will trip someone up. :bike::house: